### ISSUE TITLE
jobstart(): allow custom environment to be passed to the job

### ISSUE THREAD

**Issue by justinmk** [neovim/neovim#4970]
libuv's `uv_spawn` allows a custom environment to be passed to the process. The `{opt}` parameter of `jobstart()` should take a `opt.env` dictionary which is mapped to the environment parameter of uv_spawn.

See also https://github.com/neovim/lua-client/pull/29

**Comment by ZyX-I** [comment]
I would say that in addition to passing custom environment, it should be possible to pass *addition* to the current environment. E.g. in https://github.com/neovim/neovim/pull/4964/commits/6626786338a9a6bc8780c0c1f35403bf1e40e559 I would rather not write

```Lua
      for _, k in ipairs({
        'HOME',
        'ASAN_OPTIONS',
        'LD_LIBRARY_PATH', 'PATH',
        'NVIM_LOG_FILE',
      }) do
        env_tbl[k] = os.getenv(k)
      end
```

, but simply take the whole existing environment and pass it down with my additions, but there is no such option. I guess most usable solution is adding a `Object api_vim_environ(Object varname)` method which takes either a String or NIL and returns either a String or Dictionary (though maybe better have `String vim_get_env(String varname)`and `Dictionary vim_get_whole_env(void)`, statically typed languages are better with this variant). This will provide both VimL function that satisfies this requirement (#4934) and a more convenient way to get environment then eval.

**Comment by lewis6991** [comment]
Closed with #7202
